pkg/apis/core/validation/validation.go

```diff
@@ -7067,31 +7067,60 @@ func validateNodeInclusionPolicy(fldPath *field.Path, policy *core.NodeInclusion
 	return nil
 }
 
-// validateMatchLabelKeys tests that the elements are a valid label name and are not already included in labelSelector.
-func validateMatchLabelKeys(fldPath *field.Path, matchLabelKeys []string, labelSelector *metav1.LabelSelector) field.ErrorList {
-	if len(matchLabelKeys) == 0 {
-		return nil
-	}
-
+// validateMatchLabelKeysAndMismatchLabelKeys checks if both matchLabelKeys and mismatchLabelKeys are valid.
+func validateMatchLabelKeysAndMismatchLabelKeys(fldPath *field.Path, matchLabelKeys, mismatchLabelKeys []string, labelSelector *metav1.LabelSelector) field.ErrorList {
 	var allErrs field.ErrorList
-	labelSelectorKeys := sets.String{}
+	allErrs = append(allErrs, validateLabelKeysWithSelector(fldPath.Child("matchLabelKeys"), "matchLabelKeys", matchLabelKeys, labelSelector)...)
+	allErrs = append(allErrs, validateLabelKeysWithSelector(fldPath.Child("mismatchLabelKeys"), "mismatchLabelKeys", mismatchLabelKeys, labelSelector)...)
 
-	if labelSelector != nil {
-		for key := range labelSelector.MatchLabels {
-			labelSelectorKeys.Insert(key)
+	matchLabelKeysSet := sets.New(matchLabelKeys...)
+	mismatchLabelKeysSet := sets.New(mismatchLabelKeys...)
+	for i, k := range matchLabelKeys {
+		if mismatchLabelKeysSet.Has(k) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("matchLabelKeys").Index(i), k, "exists in both matchLabelKeys and mismatchLabelKeys"))
 		}
-		for _, matchExpression := range labelSelector.MatchExpressions {
-			labelSelectorKeys.Insert(matchExpression.Key)
+	}
+	for i, k := range mismatchLabelKeys {
+		if matchLabelKeysSet.Has(k) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("mismatchLabelKeys").Index(i), k, "exists in both matchLabelKeys and mismatchLabelKeys"))
```

Comment:
This is redundant as L7078~L7082 have already covered all duplicated cases. And L7078 can be updated to iterate from raw `matchLabelKeys`.

Reply:
This part was intended to have two error messages (one for matchLabelKeys, another for mismatchLabelKeys) when the key is duplicated in both matchLabelKeys and mismatchLabelKeys. Do you mean we only need to have one error message for matchLabelKeys in such case?

Reply:
> Do you mean we only need to have one error message for matchLabelKeys in such case?

Yes, like: X = [1,2], Y=[2,3], it's good enough to error out that "2 is duplicated in X and Y". (instead of saying "2 is duplicated in X with Y; 2 is duplicated in Y with X"

Reply:
Sure. Fixed.